-------------------------------------------------------------------------------
Test set: CloudContainers.CucumberTest
-------------------------------------------------------------------------------
Tests run: 48, Failures: 23, Errors: 5, Skipped: 0, Time elapsed: 0.845 sec <<< FAILURE!
New client succesfully added(Adding a client to register)  Time elapsed: 0.012 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.display_success_message(StepDefinition.java:38)
	at ?.display success message(file:src/main/resources/AddClient.feature:26)

Client already exists(Adding a client to register)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.existing_client(StepDefinition.java:45)
	at ?.existing client(file:src/main/resources/AddClient.feature:31)

Missing parameter(Adding a client to register)  Time elapsed: 0.027 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[Client was successfully added]> but was:<[There is a missing parameter]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.missing_parameter_error_message_is_thrown(StepDefinition.java:66)
	at ?.missing parameter error message is thrown(file:src/main/resources/AddClient.feature:39)

Email is invalid(Adding a client to register)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[There is a missing parameter]> but was:<[wrongEmail is not a valid email]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.invalid_email_error_is_displayed_for(StepDefinition.java:82)
	at ?.invalid email error is displayed for "wrongEmail"(file:src/main/resources/AddClient.feature:45)

Birthdate is invalid(Adding a client to register)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[There is a missing parameter]> but was:<[111-02-1992 is not a valid birthdate]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.invalid_birtdate_error_is_displayed_for(StepDefinition.java:92)
	at ?.invalid birtdate error is displayed for "111-02-1992"(file:src/main/resources/AddClient.feature:51)

access existing data(access data as client)  Time elapsed: 0.015 sec  <<< ERROR!
java.lang.NullPointerException
	at CloudContainers.StepDefinition.the_status_of_the_container_is_returned(StepDefinition.java:689)
	at ?.the status of the container is returned(file:src/main/resources/accessData.feature:30)

journey has not started(access data as client)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[You don't own this containe]r> but was:<[Ship's still at harbou]r>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.a_error_message_is_returned_for_journey_has_not_started(StepDefinition.java:708)
	at ?.a error message is returned for journey has not started(file:src/main/resources/accessData.feature:52)

logistic company wants to see history of container(A container has a history log of its journeys, clients and status data)  Time elapsed: 0.003 sec  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at CloudContainers.StepDefinition.an_array_containing_pairs_of_journeyIDs_and_clientIDs_are_returned(StepDefinition.java:737)
	at ?.an array containing pairs of journeyIDs and clientIDs are returned(file:src/main/resources/containerHistory.feature:29)

client wants to see history of container owned(A container has a history log of its journeys, clients and status data)  Time elapsed: 0.002 sec  <<< ERROR!
java.lang.NullPointerException
	at CloudContainers.StepDefinition.an_array_containing_relevant_journeys_are_returned(StepDefinition.java:762)
	at ?.an array containing relevant journeys are returned(file:src/main/resources/containerHistory.feature:47)

client wants to see history of container but container does not exist(A container has a history log of its journeys, clients and status data)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<C[lient] does not exist> but was:<C[ontainer] does not exist>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.response_message_saying_that_container_does_not_exist_to_client(StepDefinition.java:772)
	at ?.response message saying that container does not exist to client(file:src/main/resources/containerHistory.feature:56)

client wants to see history of container owned with no journeys(A container has a history log of its journeys, clients and status data)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NullPointerException
	at CloudContainers.StepDefinition.an_array_containing_relevant_journeys_are_returned_with_size(StepDefinition.java:777)
	at ?.an array containing relevant journeys are returned with size 0(file:src/main/resources/containerHistory.feature:64)

Successfully add a container to journey(Add container to journey)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_registered_client(StepDefinition.java:356)
	at ?.a logistic company with a registered client(file:src/main/resources/containerOnJourney.feature:24)

Container does not exist(Add container to journey)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_registered_client(StepDefinition.java:356)
	at ?.a logistic company with a registered client(file:src/main/resources/containerOnJourney.feature:40)

Journey does not exist(Add container to journey)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_registered_client(StepDefinition.java:356)
	at ?.a logistic company with a registered client(file:src/main/resources/containerOnJourney.feature:48)

succesfull allocation(Allocate a container to a client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_client(StepDefinition.java:267)
	at ?.a logistic company with a client(file:src/main/resources/containerToClient.feature:24)

container already owned(Allocate a container to a client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_another_client(StepDefinition.java:299)
	at ?.a logistic company with another client(file:src/main/resources/containerToClient.feature:31)

container does not exist(Allocate a container to a client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_client(StepDefinition.java:267)
	at ?.a logistic company with a client(file:src/main/resources/containerToClient.feature:38)

a journey successfully ended(End a journey)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<...successfully ended. [0] containers were set...> but was:<...successfully ended. [1] containers were set...>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.message_displayed_saying_journey_successfully_ended_for_containers(StepDefinition.java:533)
	at ?.message displayed saying journey successfully ended for 1 containers(file:src/main/resources/endJourney.feature:27)

journey does not exist(End a journey)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[Client] does not exist> but was:<[Journey] does not exist>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.a_container_allocated_to_a_client_is_added_to_the_journey_pt2(StepDefinition.java:550)
	at ?.a container allocated to a client is added to the journey pt2(file:src/main/resources/endJourney.feature:32)

a container is successfully freed(Free a container)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[This container does not belong to a client]> but was:<[Container was successfully freed]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.a_succes_message_is_displayed_for_freeing_a_container(StepDefinition.java:579)
	at ?.A succes message is displayed for freeing a container(file:src/main/resources/freeContainer.feature:28)

container is on journey(Free a container)  Time elapsed: 0.001 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<This container [does not belong to a client]> but was:<This container [is on a journey]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.a_error_message_is_displayed_that_container_is_on_journey(StepDefinition.java:588)
	at ?.A error message is displayed that container is on journey(file:src/main/resources/freeContainer.feature:37)

A logistic company wants to remove a client(Remove client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_with_a_client_with_clientID(StepDefinition.java:210)
	at ?.A logistic company that has a client with clientID 1(file:src/main/resources/removeClient.feature:24)

A logistic company wants to remove a client(Remove client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.a_logistic_company_that_has_a_client_with_an_email(StepDefinition.java:221)
	at ?.A logistic company that has a client with an email "j1@gmail.com"(file:src/main/resources/removeClient.feature:30)

Elapse time of journey for ten hours(elapse time for journey)  Time elapsed: 0.002 sec  <<< ERROR!
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:458)
	at CloudContainers.StepDefinition.data_has_been_collected_up_till_hours(StepDefinition.java:645)
	at ?.data has been collected up till 10 hours(file:src/main/resources/runJourney.feature:30)

journey ends(elapse time for journey)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:458)
	at CloudContainers.StepDefinition.data_has_been_collected_up_till_hours(StepDefinition.java:645)
	at ?.data has been collected up till 5 hours(file:src/main/resources/runJourney.feature:41)

Searching for an existing client(Logistic company search for client)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.the_client_is_returned_for_and_a_succes_message_is_displayed(StepDefinition.java:151)
	at ?.The client is returned for "email123@mail.dk" and a succes message is displayed(file:src/main/resources/searchClient.feature:26)

Searching for an existing client(Logistic company search for client)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at CloudContainers.StepDefinition.the_client_is_returned_for_and_a_succes_message_is_displayed(StepDefinition.java:171)
	at ?.The client is returned for 1 and a succes message is displayed(file:src/main/resources/searchClient.feature:33)

Client wants to update email(Update client information)  Time elapsed: 0.002 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[Non-existing client]> but was:<[Email has been updated]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at CloudContainers.StepDefinition.display_email_update_success_message(StepDefinition.java:112)
	at ?.Display email update success message(file:src/main/resources/updateClient.feature:26)

